Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -0,0 +1,171 @@
+                    this.OnRequest(evnt.Key, evnt.Value, currentActivity);
+                    break;
+                case "Microsoft.Azure.ServiceBus.Exception":
+                    this.OnException(evnt.Key, evnt.Value);
+                    break;
+                default:
+                    this.OnDependency(evnt.Key, evnt.Value, currentActivity);
+                    break;
+            }
+        }
+
+        private void OnDependency(string name, object payload, Activity activity)
+        {
+            DependencyTelemetry telemetry = new DependencyTelemetry { Type = DependencyType };
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+
+            PropertyFetcher urlFetcher = this.GetOrCreatePropertyFetcher(name, EndpointPropertyName);
+
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            telemetry.Data = ((Uri)urlFetcher.Fetch(payload)).ToString();
+
+            PropertyFetcher entityFetcher = this.GetOrCreatePropertyFetcher(name, EntityPropertyName);
+
+            // Queue/Topic name, e.g. myqueue/mytopic
+            telemetry.Target = (string)entityFetcher.Fetch(payload);
+
+            this.telemetryClient.TrackDependency(telemetry);
+        }
+
+        private void OnRequest(string name, object payload, Activity activity)
+        {
+            RequestTelemetry telemetry = new RequestTelemetry();
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+
+            PropertyFetcher urlFetcher = this.GetOrCreatePropertyFetcher(name, EndpointPropertyName);
+
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            telemetry.Url = (Uri)urlFetcher.Fetch(payload);
+
+            PropertyFetcher entityFetcher = this.GetOrCreatePropertyFetcher(name, EntityPropertyName);
+            
+            // We want to make Source field extendable at the beginning as we may add
+            // multi ikey support and also build some special UI for requests coming from the queues.
+            telemetry.Source = "roleName:" + (string)entityFetcher.Fetch(payload);
+
+            this.telemetryClient.TrackRequest(telemetry);
+        }
+
+        private void OnException(string name, object payload)
+        {
+            PropertyFetcher exceptionFetcher = this.GetOrCreatePropertyFetcher(name, ExceptionPropertyName);
+            Exception ex = (Exception)exceptionFetcher.Fetch(payload);
+
+            this.telemetryClient.TrackException(ex);
+        }
+
+        private void SetCommonProperties(string eventName, object eventPayload, Activity activity, OperationTelemetry telemetry)
+        {
+            // activity name looks like 'Microsoft.Azure.ServiceBus.<Name>'
+            // as namespace is too verbose, we'll just take the last node from the activity name as telemetry name
+            string[] activityNameSegments = activity.OperationName.Split('.');
```

Comment:
`string.Split` is expensive. If all you need is the part after the last period, use `LastIndexOf` to find the last period and `Substring` to extract everything after that.

Reply:
good point